src/ray/raylet/node_manager.cc

```diff
@@ -286,7 +287,7 @@ void NodeManager::ProcessNewClient(LocalClientConnection &client) {
 
 void NodeManager::DispatchTasks() {
   // Work with a copy of scheduled tasks.
-  auto scheduled_tasks = local_queues_.GetScheduledTasks();
+  const auto& scheduled_tasks = local_queues_.GetScheduledTasks();
```

Comment:
no, here I explicitly want to work with a copy (as per the comment on L290), because we're calling `RemoveTasks` on local queues below, which will change the contents of `scheduled_tasks_`. This will modify a container we're iterating over, if it's a const ref.

Reply:
got your point, will rollback it.

Reply:
How about change the following change to 

std::vector<TaskID> dispatched_taskIDs;
dispatched_taskIDs.reserve(scheduled_tasks.size());

 const auto &local_resources =
        cluster_resource_map_[my_client_id].GetAvailableResources();
for (const auto &task : scheduled_tasks) {
    const auto &task_resources = task.GetTaskSpecification().GetRequiredResources();
    if (!task_resources.IsSubset(local_resources)) {
      // Not enough local resources for this task right now, skip this task.
      continue;
    }
  dispatched_taskIDs.push_back(task.GetTaskSpecification().TaskId());
}
 // We have enough resources for this task. Assign task.
 // TODO(atumanov): perform the task state/queue transition inside AssignTask.
  auto dispatched_tasks =
        local_queues_.RemoveTasks(dispatched_taskIDs);
for (auto& task : dispatched_tasks)
    AssignTask(task);

Reply:
In this way, we don't need copy scheduled_tasks, and only need call the RemoveTasks for once.